the-collab-lab/tcl-45-smart-shopping-list
# Tr edit item

## Description

This PR adds a new feature giving users the ability to edit an item on their list. This allows uses to edit the text of an item without the need to delete the item and re-create it. The benefit of this is that any previous purchase history that has been established for that particular item will not be affected or reset. 

In `ListItem.jsx` a variable called `itemDisplay` was initialized. A new piece of state called `isEditing` was created. When a user clicks the edit button, `isEditing` is set to 'true'. While `isEditing` is true, a form input displays allowing the user to edit the name of the item. On submission of the form, a function in `firebase.js` named `updateItemName` is called and `isEditing` is set back to false. This function updates the item name in the Firestore database. While `isEditing` is false, `itemDisplay` shows the purchase check box, the item name label, and the edit button. 

## Related Issue

Closes #51 

## Acceptance Criteria
- [x] Allow users to edit any given item on their list
- [x] Editing an item also updates the item name in Firebase

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

![Screen Shot 2022-08-26 at 7 54 02 AM](https://user-images.githubusercontent.com/90880701/186933058-e85bec47-4176-4407-930f-c3f36de93da4.png)

![Screen Shot 2022-08-26 at 7 56 17 AM](https://user-images.githubusercontent.com/90880701/186933725-ad0fbf1f-f425-4897-a508-9834df32805e.png)

### After

![Screen Shot 2022-08-26 at 7 54 36 AM](https://user-images.githubusercontent.com/90880701/186933296-67d94c33-3550-4530-9173-88286a355be5.png)

![Screen Shot 2022-08-26 at 7 55 06 AM](https://user-images.githubusercontent.com/90880701/186933450-273a489b-3783-4834-80d5-e4e0521824ec.png)

![Screen Shot 2022-08-26 at 7 55 28 AM](https://user-images.githubusercontent.com/90880701/186933536-363f2ae7-1151-45bd-a581-2f15396e860e.png)

![Screen Shot 2022-08-26 at 7 56 46 AM](https://user-images.githubusercontent.com/90880701/186933809-d3289ad4-5281-4b9a-bbe1-3d507d637f99.png)

## Testing Steps / QA Criteria

1. From your local machine, `git checkout main` then `git pull`
2. `git checkout TR-edit-item`
3. `npm start`
4. If you have not already created a list, create one & start adding items.
5. In a separate browser window (recommend having it to the side of the app) open the Firestore database
6. Edit an item on the list. You should see, in real time, the Firestore database update with the newly edited name, as well as see the update displayed on the Front End of the app
